modules/data_models.py

```diff
@@ -0,0 +1,363 @@
+
+            max_arg2 = {"value": day_data.get_max_humidity(), "date": date}
+            max_humidity = max(
+                max_humidity, max_arg2, key=lambda x: x["value"] or -1000
+            )
+        return [
+            max_temperature,
+            min_temperature,
+            max_humidity,
+        ]
+
+    def get_averages(self):
+        """
+        Returns the averages highest_temperature, lowest_temperature and mean_humidity
+        Returns:
+            (list or None): list containing rounded averages
+                            [
+                                avg_highest_temperature(str),
+                                avg_lowest_temperature(str),
+                                avg_mean_humidity(str)
+                            ]
+                            Or
+                            None if no data exists for the month
+        """
+        if not self.days_data:
+            return None
+        sum_highest_temperature = 0
+        sum_lowest_temperature = 0
+        sum_mean_humidity = 0
+        for day_data in self.days_data:
+            sum_highest_temperature += day_data.get_max_temperature() or 0
+
+            sum_lowest_temperature += day_data.get_min_temperature() or 0
+
+            sum_mean_humidity += day_data.get_mean_humidity() or 0
+
+        return [
+            f"{round(sum_highest_temperature / len(self.days_data))}{TEMPERATURE_UNIT}",
+            f"{round(sum_lowest_temperature / len(self.days_data))}{TEMPERATURE_UNIT}",
+            f"{round(sum_mean_humidity / len(self.days_data))}{HUMIDITY_UNIT}",
+        ]
+
+    def get_month_values(self):
+        """
+        Returns date, highest temperature, lowest temperature and max humidity for each day
+        Returns:
+            (list or None): list containing year, month, day, max_temperature and min_temperature
+                            for each day of the month
+                            month_data[i] = [
+                                year(str),
+                                name(str),
+                                day(str),
+                                max_temperature_with_unit(str),
+                                min_temperature_with_unit(str)
+                            ]
+        """
+
+        if not self.days_data:
+            return None
+        month_data = []
+        for day_data in self.days_data:
+            month_data.append(
+                [
+                    self.year,
+                    self.name,
+                    day_data.get_date().split("-")[2],
+                    day_data.get_max_temperature(),
+                    day_data.get_min_temperature(),
+                ]
+            )
+
+        return month_data
+
+    def generate_averages_report_month(self):
+        """
+        Generates report using the averages list and displays
+        1. average highest temperature
+        2. average lowest temperature
+        3. average mean humidity
+        for a month
```

Comment:
This should be in generator class i suppose, once you have the required data from the month you can pass that yo generator class for further processing.

Reply:
Done